samples/subsys/usb/mass/src/main.c

```diff
@@ -125,3 +115,18 @@ void main(void)
 	flash_area_close(pfa);
 #endif
 }
+
+void main(void)
+{
+	int ret;
+
+	setup_flash();
```

Comment:
``` diff
diff --git a/samples/subsys/usb/mass/src/main.c b/samples/subsys/usb/mass/src/main.c
index a021f2befa..320b56f33e 100644
--- a/samples/subsys/usb/mass/src/main.c
+++ b/samples/subsys/usb/mass/src/main.c
@@ -47,7 +47,6 @@ static struct fs_mount_t fs_mnt = {
 
 static void setup_flash(void)
 {
-#if CONFIG_DISK_ACCESS_FLASH
 	struct fs_mount_t *mp = &fs_mnt;
 	unsigned int id = (uintptr_t)mp->storage_dev;
 	int rc;
@@ -113,14 +112,14 @@ static void setup_flash(void)
 
 out:
 	flash_area_close(pfa);
-#endif
 }
 
 void main(void)
 {
 	int ret;
-
+#if IS_ENABLED(CONFIG_DISK_ACCESS_FLASH)
 	setup_flash();
+#endif
 
 	ret = usb_enable(NULL);
 	if (ret != 0) {
```

Reply:
You'd get an "unreferenced function" warning when `CONFIG_DISK_ACCESS_FLASH` is not selected.

Reply:
Hm. You are right, I forgot it. So maybe it would be better to put definition of this function inside `CONFIG_DISK_ACCESS_FLASH` ifdefs as well, not only the function body? I just don't like the fact that at first sight You can see that `setup_flash()` is called even when You build with `overlay-ram-disk.conf`

Reply:
What if I rename it `setup_disk()`?

Reply:
Could be as well. I will just post alternative solution and let other speak.
``` diff
diff --git a/samples/subsys/usb/mass/src/main.c b/samples/subsys/usb/mass/src/main.c
index a021f2befa..76ba1dc42f 100644
--- a/samples/subsys/usb/mass/src/main.c
+++ b/samples/subsys/usb/mass/src/main.c
@@ -43,11 +43,8 @@ static struct fs_mount_t fs_mnt = {
 #else
 #error No recognized file system
 #endif  /* file system */
-#endif  /* CONFIG_DISK_ACCESS_FLASH */
-
 static void setup_flash(void)
 {
-#if CONFIG_DISK_ACCESS_FLASH
 	struct fs_mount_t *mp = &fs_mnt;
 	unsigned int id = (uintptr_t)mp->storage_dev;
 	int rc;
@@ -113,14 +110,16 @@ static void setup_flash(void)
 
 out:
 	flash_area_close(pfa);
-#endif
 }
+#endif  /* CONFIG_DISK_ACCESS_FLASH */
 
 void main(void)
 {
 	int ret;
 
+#if IS_ENABLED(CONFIG_DISK_ACCESS_FLASH)
 	setup_flash();
+#endif
 
 	ret = usb_enable(NULL);
 	if (ret != 0) {
```